src/batchUploader.ts

```diff
@@ -245,6 +333,14 @@ export class BatchUploader {
             this.batchesQueuedForProcessing.unshift(...batchesThatDidNotUpload);
         }
 
+        // Update Offline Storage with current state of batch queue
+        if (this.offlineStorageEnabled && this.batchVault) {
+            this.batchVault.store(this.batchesQueuedForProcessing);
+
+            // Clear batch queue since everything should be in Offline Storage
+            this.batchesQueuedForProcessing = [];
+        }
```

Comment:
I wonder if this method for queueing events can be split out to be easier to grok (even if this is a bit less DRY).  This method has a lot of if `offlineStorageEnabled` logic, and its a bit hard to keep track of what is stored when with this long if statement.  Perhaps one path should be `if offlineStorageEnabled` and the other path be if (!offlineStorageEnabled) (or what is the current process).  This way we are separating the concerns more, as opposed to having to think about what state we are in if `offlineStorageEnabled` is true/false.

Reply:
Agreed but I think it might expand the scope of this initiative. I know we've spoken about decoupling batch queuing from uploading, so perhaps it's a sustained engineering project?